File: etl/models/despesa_candidato.py
from sqlalchemy import Column, BigInteger, String, ForeignKey, Numeric, Date, Text, UniqueConstraint
from .base import Base

class DespesaCandidato(Base):
    __tablename__ = "despesa_candidato"

    id_despesa = Column(BigInteger, primary_key=True, autoincrement=True)
    sq_candidato = Column(BigInteger, ForeignKey("candidatura.sq_candidato"), nullable=False)
    nr_documento = Column(String)
    cpf_cnpj_fornecedor = Column(String)
    nm_fornecedor = Column(String)
    dt_despesa = Column(Date)
    vr_despesa = Column(Numeric(15, 2))
    ds_tipo_despesa = Column(String)
    ds_fonte_recurso = Column(String)
    ds_especie_recurso = Column(String)
    ds_despesa = Column(Text)

    __table_args__ = (
        UniqueConstraint("sq_candidato", "nr_documento", "cpf_cnpj_fornecedor", "dt_despesa"),
    )

File: src/backend/VoteBem/Entities/DespesaCandidato.cs
namespace VoteBem.Entities
{
    public class DespesaCandidato
    {
        public long IdDespesa { get; private set; }
        public long SqCandidato { get; private set; }
        public string? NrDocumento { get; private set; }
        public string? CpfCnpjFornecedor { get; private set; }
        public string? NmFornecedor { get; private set; }
        public DateOnly? DtDespesa { get; private set; }
        public decimal? VrDespesa { get; private set; }
        public string? DsTipoDespesa { get; private set; }
        public string? DsFonteRecurso { get; private set; }
        public string? DsEspecieRecurso { get; private set; }
        public string? DsDespesa { get; private set; }

        public Candidatura Candidatura { get; private set; } = null!;

        protected DespesaCandidato() { }
    }
}
